### PR TITLE
Update Spacemacs installation instructions on Readme.md

### LINKED ISSUE #12: Layer not automatically opening for vue files
I've installed vue-mode to spacemacs like so, but it doesn't automatically enable when I open a `.vue` file. I have to manually `M-x vue-mode` to get it to work. Do I need to do anything extra to get it to autoload?

```el
     (vue-mode :location (recipe
                          :fetcher github
                          :repo "codefalling/vue-mode"))
```

### LINKED ISSUE #24: Readme is very unfriendly for non-spacemacs experts
The readme needs a little work; I was just a witness to this conversation:

image

If your package is the reason people abandon spacemacs, that's a problem :/

### PR DESCRIPTION
**Pull request by jaysoifer**
You might want to squash the commits 😄 
Closes #24 
Also apparently Closes #12 (at least the original question)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,31 +20,76 @@ Emacs users may install the package from MELPA, and then add the following to th
...
+   • So the section should look like this, after the update:
+   ```lisp
+   dotspacemacs-additional-packages '(
+   (vue-mode :location (recipe
+                        :fetcher github
+                        :repo "codefalling/vue-mode"))
+   )
+   ```
+   5. Locate, in the same file, the section named `defun dotspacemacs/user-config ()`
+   6. The following code should code right in that section:
```
Comment (lazywithclass) [inline README.md]: @jaysoifer small typo here, I think you meant "The following code should *go*".

File: README.md [hunk 2]
```diff
@@ -20,31 +20,76 @@ Emacs users may install the package from MELPA, and then add the following to th
...
+   1. Add to your `packages`:
+   ```lisp
+   (vue-mode :location (recipe
+                        :fetcher github
+                        :repo "codefalling/vue-mode"))
+   ```
+   2. Add to your `package.el`:
+
+   ```lisp
+   (defun your-layer-name/init-vue-mode ()
```
Comment (lazywithclass) [inline README.md]: @jaysoifer who is calling this function? Is this a way to have Spacemacs call "init" function, like calling them "init-$mode-name"?

**Comment by lazywithclass** [comment]
This was very useful for me, thanks! I've added a couple comments.

File: README.md [hunk 3]
```diff
@@ -20,31 +20,76 @@ Emacs users may install the package from MELPA, and then add the following to th
...
+   • So the section should look like this, after the update:
+   ```lisp
+   (defun dotspacemacs/user-config ()
+   "Configuration function for user code.
+   This function is called at the very end of Spacemacs initialization after
+   layers configuration.
+   This is the place where most of your configurations should be done. Unless it is
+   explicitly specified that a variable should be set before a package is loaded,
+   you should place your code here."
+   (defun dotspacemacs/init-vue-mode ()
```
Comment (lazywithclass) [inline README.md]: @jaysoifer who is calling this function? Is this a way to have Spacemacs call "init" functions, like calling them "init-$mode-name"?

File: README.md (hunk 1 above)
Comment (jaysoifer) [inline README.md]: Fixing it, thank you :)

File: README.md [hunk 4]
```diff
@@ -20,31 +20,76 @@ Emacs users may install the package from MELPA, and then add the following to th
...
+                        :fetcher github
+                        :repo "codefalling/vue-mode"))
+   ```
+   • So the section should look like this, after the update:
+   ```lisp
+   dotspacemacs-additional-packages '(
+   (vue-mode :location (recipe
+                        :fetcher github
+                        :repo "codefalling/vue-mode"))
+   )
```
Comment (AdamNiederer) [inline README.md]: Small nitpick: Could we get the entire s-expression (or at least the relevant bit) in this block? Having the whole thing including `(defun dotspacemacs/layers ()` might make it a little bit easier for people skimming the docs

**Comment by AdamNiederer** [comment]
This looks fantastic, thanks so much for your work on this!

I've outlined a few tiny things above; the other one that sticks out is just following emacs' coding style rules for ending braces (just dump them all on the same line). LGTM otherwise 👍

File: README.md (hunk 1 above)
Comment (jaysoifer) [inline README.md]: Yes, on to it :)

File: README.md (hunk 3 above)
Comment (jaysoifer) [inline README.md]: > `dotspacemacs/user-config` is called at the very end of Spacemacs initialization after layers configuration. This is the place where most of your configurations should be done. Unless it is explicitly specified that a variable should be set before a package is loaded, you should place your code here.

[Source](http://spacemacs.org/doc/DOCUMENTATION.html#dotfile-contents)

File: README.md (hunk 4 above)
Comment (jaysoifer) [inline README.md]: I'm not sure what you mean by this...
You want me to remove the "encompassing" block or add something else?

File: README.md (hunk 3 above)
Comment (jaysoifer) [inline README.md]: This is as far as I can go 😓 ... I don't know the answer to your question 😊

**Comment by AdamNiederer** [comment]
Yep; looks good.

**Pull request merged**